builder/frameworks/component_manager.py

```
@@ -835,195 +857,289 @@
             return
 
+        if not self.config.arduino_libs_mcu:
+            return
+
         build_py_path = str(Path(self.config.arduino_libs_mcu) / "pioarduino-build.py")
         backup_path = str(Path(self.config.arduino_libs_mcu) / f"pioarduino-build.py.{self.config.mcu}")
 
         if os.path.exists(build_py_path) and not os.path.exists(backup_path):
             shutil.copy2(build_py_path, backup_path)
 
+    def restore_pioarduino_build_py(self, target=None, source=None, env=None) -> None:
```

Ruff (0.12.2)

Unused method argument: `target`

(ARG002)